test/channels-cornercases.c

```diff
@@ -120,7 +120,7 @@ int main() {
 	// Start MeshLink and wait for the channel to become connected.
 	start_meshlink_pair(a, b);
 
-	assert(wait_sync_flag(&channel_opened, 20));
+	assert(wait_sync_flag_ex(&channel_opened, 20));
```

Comment:
We should never have to call `wait_sync_flag_ex()`. That means we are holding the mutex, but the mutex should only be touched by the `*_sync_flag()` functions. The problem is that we didn't lock it correctly before, and this was fixed by adding `pthread_mutex_lock(&channel_opened.mutex)` in `channel-cornercases.c`. However, that call to `pthread_mutex_lock()` should be removed, and then we can just call `wait_sync_flag()` here. Or am I missing something?